packages/select/src/types.ts

```diff
@@ -66,15 +76,34 @@ export type RematchSelect<M extends object = Models, RootState = any> = ((
 	StoreSelectors<RootState>
 
 declare module '@rematch/core' {
-	interface Model<
+	export interface Model<
 		TModels extends Models<TModels> = Record<string, any>,
 		TState = any,
 		TBaseState = TState
 	> {
+		name?: string
+		state: TState
 		selectors?: ModelSelectorsConfig<TState>
+		reducers?: ModelReducers<TState>
+		baseReducer?: ReduxReducer<TBaseState>
+		effects?: ModelEffects<TModels> | ModelEffectsCreator<TModels>
+	}
+
+	export interface ModelResult<S, R, BR, E> {
+		name?: string
+		state: S
+		reducers?: R
+		selectors?: ModelSelectorsConfig<S>
+		baseReducer?: BR
+		effects?: E
 	}
 
-	interface RematchStore<TModels> {
-		select: RematchSelect<TModels, RematchRootState<TModels>>
+	export interface RematchStore<
```

Comment:
why copy the same RematchStore in select?

Reply:
i forgot to add the `RematchSelect` type, that's why it's copied

`		select: RematchSelect<TModels, RematchRootState<TModels>>	`

Reply:
when select plugin it's inserted the store contains a new prop `store.select()`

```jsx
const selection = store.select(models => ({
  // todo: this isn't getting autocompleted
  total: models.cart.total,
}))

const mapState = (state: RootState) => ({
	settingsState: state.settings,
  loadingState: state.loading,
  playersState: state.players,
  // todo: 2n param props shouldn't be required
  ...selection(state, null)
})
```

Reply:
Sounds nice to me